app/views/shared/_flash_message.html.erb

```diff
@@ -1,13 +1,17 @@
 <% [:error, :success, :warning, :notice].each do |key| -%>
-  <% unless flash[key].nil? -%>
-    <% if flash[key].is_a?(Array) -%>
-      <% flash[key].each do |msg| -%>
-        <%= content_tag :div, msg, class: key %>
-        <% response.header['X-message ' + key.to_s] = msg %>
+  <% if flash[key].is_a?(Array) -%>
+    <% flash[key].each do |msg| -%>
+      <% unless msg -%>
+        <%= content_tag :div, msg, class: "#{key} flash-#{key}", style: "display: none" %>
```

Comment:
Also, I would much prefer having one div per flash type, and if there are multiple messages, they appear as an unordered list within this one div. If you could fix this while you're at it, that would be much appreciated.